app/main.py

```diff
@@ -1,8 +1,25 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for person in people:
+        person_temp = Person(person["name"], person["age"])
+        person_list.append(person_temp)
+
+    for person in people:
+        if "wife" in person:
+            if person["wife"] is not None:
```

Comment:
You could avoid nesting and use `get` to perform the check
```suggestion
        if person.get("wife"):
```
Besides that, you could save the retrieved value to a variable using walrus operator as well
```suggestion
        if wife_name := person.get("wife"):
```